test/Microsoft.AspNetCore.Mvc.Razor.Test/Internal/ViewComponentTagHelperDescriptorFactoryTest.cs

```diff
@@ -87,6 +86,13 @@ public static TheoryData TypeData
         [MemberData(nameof(TypeData))]
         public void GetCSharpTypeName_ReturnsCorrectTypeNames(Type type, string expected)
         {
+            // Arrange
+            // Work around problems serializing / deserializing this ref type.
+            if (type == null)
+            {
+                type = typeof(ViewComponentTagHelperDescriptorFactoryTest).GetMethod("MethodWithOutParam").GetParameters().First().ParameterType;
+            }
```

Comment:
BTW I suspect the issue is actually in serialization -- ends up as an open generic `List<>` type.